src/m365/pp/commands/tenant/tenant-settings-set.spec.ts

```diff
@@ -0,0 +1,438 @@
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the enableOpenAiBotPublishing option is set to true', async () => {
+    const actual = await command.validate({ options: { enableOpenAiBotPublishing: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the enableModelDataSharing option is set to false', async () => {
+    const actual = await command.validate({ options: { enableModelDataSharing: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the enableModelDataSharing option is set to true', async () => {
+    const actual = await command.validate({ options: { enableModelDataSharing: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('successfully updates tenant settings', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === "https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/scopes/admin/updateTenantSettings?api-version=2020-10-01") {
+        return successResponse;
+      }
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        walkMeOptOut: false,
+        disableNPSCommentsReachout: false,
+        disableNewsletterSendout: false,
+        disableEnvironmentCreationByNonAdminUsers: false,
+        disablePortalsCreationByNonAdminUsers: false,
+        disableSurveyFeedback: false,
+        disableTrialEnvironmentCreationByNonAdminUsers: false,
+        disableCapacityAllocationByEnvironmentAdmins: false,
+        disableSupportTicketsVisibleByAllUsers: false,
+        disableDocsSearch: false,
+        disableCommunitySearch: false,
+        disableBingVideoSearch: false,
+        shareWithColleaguesUserLimit: 10000,
+        disableShareWithEveryone: false,
+        enableGuestsToMake: false,
+        disableMembersIndicator: false,
+        disableMakerMatch: false,
+        disablePreferredDataLocationForTeamsEnvironment: false,
+        disableAdminDigest: false,
+        disableDeveloperEnvironmentCreationByNonAdminUsers: false,
+        disableBillingPolicyCreationByNonAdminUsers: false,
+        storageCapacityConsumptionWarningThreshold: 85,
+        disableChampionsInvitationReachout: false,
+        disableSkillsMatchInvitationReachout: false,
+        disableCopilot: false,
+        enableOpenAiBotPublishing: false,
+        enableModelDataSharing: false
+      }
+    } as any);
+    assert(loggerLogSpy.calledWith(successResponse));
+  });
+
+  it('handles error correctly', async () => {
+    const error = {
+      error: {
+        message: "The request content was invalid and could not be deserialized."
+      }
+    };
+
+    sinon.stub(request, 'post').callsFake(async () => {
+      throw error;
+    });
+
+    await assert.rejects(command.action(logger, { options: {} } as any), error.error);
```

Comment:
Currently the test is not really testing much. Error.error is an object with property message, yet the test is succeeding just fine. Let's wrap this error in a `CommandError` to actually check that we are logging the right error.